File: 0538.把二叉搜索树转换为累加树/0538-把二叉搜索树转换为累加树.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def convertBST(self, root: TreeNode) -> TreeNode:
        def dfs(root: TreeNode) :
            nonlocal accumulator
            if root == None :
                return
            
            dfs(root.right)
            accumulator += root.val
            root.val = accumulator
            dfs(root.left)
        
        accumulator = 0
        dfs(root)
        return root

        
        

File: 0538.把二叉搜索树转换为累加树/0538-把二叉搜索树转换为累加树.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func convertBST(root *TreeNode) *TreeNode {
    var dfs func(*TreeNode)
    accumulator := 0
    
    dfs = func(root *TreeNode) { 
        if root == nil {
            return
        }

        dfs(root.Right)
        accumulator  += root.Val
        root.Val = accumulator
        dfs(root.Left)
    }

    dfs(root)
    return root
}